spotbugs/src/main/java/edu/umd/cs/findbugs/TextUICommandLine.java

```diff
@@ -751,6 +755,16 @@ public void configureEngine(IFindBugsEngine findBugs) throws IOException, Filter
             findBugs.setApplySuppression(true);
         }
 
+        // Ideally applied last, to make sure if the user preferences file is
+        // given, it overrides the values set by the command line options.
+        if (parsedOptions.containsKey(USER_PREFS)) {
+            // only override values if the user preferences file was given, to avoid re-setting to defaults
+            textuiBugReporter.setPriorityThreshold(userPreferences.getUserDetectorThreshold());
+            textuiBugReporter.setRankThreshold(userPreferences.getFilterSettings().getMinRank());
+            findBugs.setAnalysisFeatureSettings(userPreferences.getAnalysisFeatureSettings());
+            findBugs.setMergeSimilarWarnings(userPreferences.getMergeSimilarWarnings());
+        }
+
```

Comment:
Maybe we should show a warning when the command line disagree with the preferences file. IMO the command line options should take over, but this is debatable and I'm happy if you wrote it the other way.
But either way, it is going to surprise some users and warning that they're was a conflict would help clarifying the behaviour.

Reply:
Added warning.